mathesar_ui/src/component-library/text-input/TextInput.svelte

```diff
@@ -18,22 +18,30 @@
   export let disabled = false;
 
   // Underlying DOM element for direct access
-  export let element: HTMLElement = null;
+  export let element: HTMLInputElement | undefined = undefined;
 
   // Id for the input
   export let id: string = undefined;
 
-  function handleKeypress(e: KeyboardEvent) {
+  export let hasValidationErrors = false;
```

Comment:
We should handle this for all input components, but we can take it up in separate PRs when needed.

Reply:
I considered that too. Certainly for components that _compose_ TextInput, like NumberInput. But I'm not sure if we need it for any other ones. The only one to _maybe_ have it for is TextArea, but I can't really think of a realistic use case there.

Reply:
Oh Oops, NumberInput doesn't compose TextInput. I was thinking of previous work I did. We might consider the usefulness of doing so.

Reply:
Ah! Shoot -- sorry -- I was moving too fast. I looked at the diff and thought you were talking about `handleKeypress`.

For `hasValidationErrors`, yes, I agree this should be applicable to TextArea for sure. For other inputs, like Checkbox, Radio, and Select I think it's probably not relevant.